include/base/auto_ptr.h

```diff
@@ -373,6 +373,13 @@ class AutoPtr
   }
 
   /**
+   * Forward compatibility with newer smart pointer types.  This
+   * allows code like if (!foo) to work with AutoPtr.
+   */
+  explicit operator bool() const
+  { return (this->get() != NULL); }
```

Comment:
You guys and your "!= NULL" pointer comparisons.  How about we return "(this->get() != NULL) == true" for good measure?  ;-)